packages/use-sync-external-store/package.json

```
@@ -17,6 +17,15 @@
     "shim/",
     "cjs/"
   ],
+  "exports": {
+    ".": "./index.js",
+    "./with-selector": "./with-selector.js",
+    "./shim": "./shim/index.js",
```

thoughts on

```suggestion
    "./shim": {
      "react-native": "./shim/index.native.js",
      "default": "./shim/index.js"
    },
```

?

That way all code can just be `use-sync-external-store/shim`.

note that this relies on the bundler actually supporting `exports` and supplying `react-native` condition (which I'm guessing `metro` at least doesn't do), but it seems more future-proof than these (seemingly) somewhat random exports

---

https://github.com/facebook/metro/issues/670
Metro doesn't respect `exports` field at all.

---

Right, so no issue.

This stanza should still work for any modern bundler, and metro whenever it enters the future

---

There's Repack to consider (basically Webpack for react-native) so we might as well get this in now. Doing it in the future might be considered another breaking change.